src/main/java/net/kaczmarzyk/spring/data/jpa/domain/IsEmpty.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2014-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.kaczmarzyk.spring.data.jpa.domain;
+
+import net.kaczmarzyk.spring.data.jpa.utils.QueryContext;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+
+
+/**
+ * <p>Filters with "is empty" where-clause for collections (e.g. {@code where customer.orders is empty}).</p>
+ *
+ * <p>Does not require any http-parameters to be present, i.e. represents constant part of the query.</p
+ *
+ * @author Hubert Gotfryd (Tratif sp. z o.o.)
+ */
+public class IsEmpty<T> extends PathSpecification<T> implements WithoutTypeConversion, ZeroArgSpecification {
+
+    private static final long serialVersionUID = 1L;
+
+    public IsEmpty(QueryContext queryContext, String path, String[] args) {
```

Comment:
`args` are not used but has to be passed because `SimpleSpecificationResolver` does not support 2-arg constructor for specifications (similar scenario in `IsNull` specification). Is is ok or maybe we should extend logic in `SimpleSpecificationResolver` (`buildSpecification` metohd).

Reply:
I'm not sure but you can try inherit by ZeroArgSpecification interface as this allows you to use contructor without third argument